src/lib.rs

```diff
@@ -183,28 +183,15 @@ mod contract {
             None => sdk::panic_utf8(b"ERR_INVALID_ECDSA_SIGNATURE"),
         };
 
-        let next_nonce =
-            Engine::check_nonce(&sender, &signed_transaction.transaction.nonce).sdk_unwrap();
```

Comment:
I'm pretty sure this is still needed because `SputnikVM` does not do any nonce checking (but it does increment the nonce on its own.

Reply:
Literally in a call with Arto where I realised this. Thanks for noticing :D

Reply:
Reminder: nonce check is still needed here. Also, if you merge in master we now have the sanity tests, which would be good to double check functionality with the changes in this PR.

Reply:
Do note that I'm making an alteration to this check. I know that it has an overflow check and in most cases that is fine, but U256 is such an inconceivably large number that it would be virtually impossible to hit this limit.